### PR TITLE
Request n

### PR DESCRIPTION
**Pull request by matthiashanel**
Adding an option to expect more than one response. (2nd commit) 
To test things I basically stopped using samples and use the nats utility only.

Had to run go mod vendor (1st commit) to get it to compile

### CODE REVIEW

**Comment by ripienaar** [comment]
Definitely do not need, or want, vendor checkin. Please undo that and let’s see what the problem is

File: nats/pub_command.go [hunk 1]
```diff
@@ -85,33 +109,46 @@ func (c *pubCmd) publish(pc *kingpin.ParseContext) error {
...
+			}
+			c.printResponse(m)
+		} else {
+			if msg.Reply == "" {
+				msg.Reply = nc.NewRespInbox()
+			}
 
-		if c.raw {
-			fmt.Println(string(m.Data))
+			msgChan := make(chan *nats.Msg, c.respCount)
```
Comment (ripienaar) [inline nats/pub_command.go:123]: really wish this whole pattern was in nats.go instead, I have several implementations of it.

Comment (matthiashanel) [reply nats/pub_command.go:123]: Agreed.

**Comment by matthiashanel** [comment]
> Definitely do not need, or want, vendor checkin. Please undo that and let’s see what the problem is

I checked out, quickly made the change and had to rebase. perhaps the missing commit fixed my issue. 
It appears to be gone now and I removed the vendor commit

**Comment by ripienaar** [comment]
@derekcollison should we look at add a Request() that has a option to wait for multiple results rather than just 1 to nats.go instead of adding more here? Would be very useful.

**Comment by derekcollison** [comment]
Sure..

File: nats/pub_command.go (hunk 1 above)
Comment (matthiashanel) [reply nats/pub_command.go:123]: Created an issue in nats.go . will update this PR once the issue is closed

**Comment by matthiashanel** [comment]
@ripienaar I won't get to adding a new api anytime soon. Until then, rebase and merge this or discard. Your call.

**Comment by ripienaar** [comment]
think I'd rather close it or do the work to extract the other similar behaviours in the nats cli into a single reusable - rather than add an additional kind of similar helper

**Pull request closed without merging**